### PR TITLE
Remove dependency on JDK 17 HexFormat

### PR DESCRIPTION
**Pull request by msgilligan**
Add ByteArray, ByteUtils, and HexFormat classes to do it.

### CODE REVIEW

File: secp-api/src/main/java/org/bitcoinj/secp/api/internal/ByteUtils.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bitcoinj.secp.api.internal;
+
+import java.util.Comparator;
+
+/**
+ *
```
Comment (schildbach) [inline secp-api/src/main/java/org/bitcoinj/secp/api/internal/ByteUtils.java:21]: An empty comment?

File: secp-api/src/main/java/org/bitcoinj/secp/api/internal/ByteUtils.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+public class ByteUtils {
+    /**
+     * Provides a byte array comparator.
+     * @return A comparator for byte[]
+     */
+    public static Comparator<byte[]> arrayUnsignedComparator() {
+        return ARRAY_UNSIGNED_COMPARATOR;
+    }
+
+    // In Java 9, this can be replaced with Arrays.compareUnsigned()
```
Comment (schildbach) [inline secp-api/src/main/java/org/bitcoinj/secp/api/internal/ByteUtils.java:32]: I assume you're planning for JDK 8? Otherweise you as per comment you could replace this.

**Review by schildbach — APPROVED** [review]
It's a bit unfortunate that this PR duplicates a couple of classes from `base.internal`. But I guess since it's internal, `secp` can't use it?

File: secp-api/src/main/java/org/bitcoinj/secp/api/internal/ByteUtils.java (hunk 2 above)
Comment (msgilligan) [reply secp-api/src/main/java/org/bitcoinj/secp/api/internal/ByteUtils.java:32]: Yes, planning for JDK 8!

File: secp-api/src/main/java/org/bitcoinj/secp/api/internal/ByteUtils.java (hunk 1 above)
Comment (msgilligan) [reply secp-api/src/main/java/org/bitcoinj/secp/api/internal/ByteUtils.java:21]: For now...

**Comment by msgilligan** [comment]
> It's a bit unfortunate that this PR duplicates a couple of classes from `base.internal`. But I guess since it's internal, `secp` can't use it?

I think the duplication is ok, for now. But I envision the following strategy:
1. Port secp-api and secp-bouncy the rest of the way back to JDK 8
2. In the bitcoinj release that is refactored to use secp-api, have `base` depend on `secp-api` and remove the duplication then.

**Comment by msgilligan** [comment]
There are definitely a few things in here that need cleanup, but I'm going to merge and we can fix them later.

**Pull request merged**